velox/exec/Window.cpp

```diff
@@ -110,6 +110,41 @@ Window::WindowFrame Window::createWindowFrame(
     }
   };
 
+  // For k Range frame bound:
+  // i) The order by needs to be a single column for bound comparisons.
+  // ii) The bounds values are pre-computed in the start(end)Value bound
+  // fields. So, start(end)Value bounds cannot be constants.
+  // iii) The frame bound column and the ORDER BY column must have
+  // the same type for correct comparisons.
+  if (frame.type == core::WindowNode::WindowType::kRange &&
+      (frame.startValue || frame.endValue)) {
+    VELOX_USER_CHECK_EQ(
+        sortKeyInfo_.size(),
+        1,
+        "Window frame of type RANGE PRECEDING or FOLLOWING requires single sort item in ORDER BY.");
+
+    auto frameBoundCheck = [&](const core::TypedExprPtr& frameValue) -> void {
+      if (frameValue) {
+        auto frameChannel = exprToChannel(frameValue.get(), inputType);
+        VELOX_USER_CHECK_NE(
+            frameChannel,
+            kConstantChannel,
+            "Window frame of type RANGE does not support constant arguments");
+
+        auto frameType = inputType->childAt(frameChannel);
+        auto orderByType = inputType->childAt(sortKeyInfo_[0].first);
```

Comment:
Can this variable be computed outside of this lambda?

Reply:
Yes. Nice catch. Thanks !